### DISCUSSION TITLE
Why and when does tasmota restart on WiFi loss

### DISCUSSION THREAD

**Discussion by dingausmwald** [arendst/Tasmota#16011] [Q&A] [answered]
I always thought tasmota runs reliable without WiFi. After setting some blitzwolf shp6 on the outer edge of my WiFi router (ca. 60% signal) I recognized frequent restarts (restart reason: software/system).

Furthermore nobody can tell me when and why tasmota restarts. Seems like even the ppl involved here can not answer this properly.

I see that ntp time sync is a thing since there is no rtc. But in my case (and I believe in most others) an inaccurate time is enough (replacing a general time switch f.e. Which isn't that accurate too).

Maybe making this an option whether tasmota restarts on WiFi loss / ntp loss or even better setting a time span (f.e. 6 hours) would be much better I think. I thought (and the docs say so btw > wificonfig) there is no restart on WiFi. 

So, could somebody shed some light on this?

**Comment by sfromis** [comment, marked as answer]
Suggesting an option would only make sense if it was shown to be a planned restart controlled by Tasmota. As I've heard of no such scheme, that leaves the issue down to debugging what happens, and how to recreate the issue.

One of my Tasmota devices is living with signal strength worse than -80 and the Tasmota version of the number down to 28. No problems with stability.

**Comment by dingausmwald** [comment]
Ah okay sorry. Thought this is tasmota initiated. Can be implied by a comment from 2.4.2022 from  you in a (still open) thermostat issue from here. 

"Point being that without Wifi, it seems that Tasmota may (not sure when) decide to restart (even with WifiConfig 4), and when unable to get a real time (from NTP server) after that, Tasmota would not work well." 

So this is a bug?! Okay. Quite difficult to debug when no WiFi is present. Only with serial, right?

Also software/system restart reason can be understood as a restart initiated by the software - tasmota. For me it's still a bit tricky to understand the restart reasons. There are no docs and from what i red so far they are all meaningless without logs (software/system, hardware watchdog, software watchdog, exception,...).

**Reply by sfromis** [reply]
Key words were "seems" and "may", as I did not find or diagnose any such scheme. What I was after was the detail that after a restart, timing breaks down without a real time available.

**Reply by pkkrusty** [reply]
You could have a system#init rule set the time, after which the system would not try to reach out to the NTP server. Would eliminate that as a cause of unexpected restarts.
